backend/src/sessionRoutes.ts

```diff
@@ -0,0 +1,133 @@
+import 'dotenv/config';
+import express from 'express';
+import session from 'express-session';
+import memoryStoreFactory from 'memorystore';
+
+import {
+	handleChatToGPT,
+	handleGetChatHistory,
+	handleAddToChatHistory,
+	handleClearChatHistory,
+} from './controller/chatController';
+import {
+	handleConfigureDefence,
+	handleDefenceActivation,
+	handleDefenceDeactivation,
+	handleGetDefenceStatus,
+	handleResetSingleDefence,
+} from './controller/defenceController';
+import {
+	handleClearEmails,
+	handleGetEmails,
+} from './controller/emailController';
+import {
+	handleConfigureModel,
+	handleGetModel,
+	handleGetValidModels,
+	handleSetModel,
+} from './controller/modelController';
+import { handleResetProgress } from './controller/resetController';
+import { ChatModel, defaultChatModel } from './models/chat';
+import { LevelState, getInitialLevelStates } from './models/level';
+
+declare module 'express-session' {
+	interface Session {
+		initialised: boolean;
+		chatModel: ChatModel;
+		levelState: LevelState[];
+	}
+}
+
+const sessionSigningSecret = process.env.SESSION_SECRET;
+if (!sessionSigningSecret) {
+	console.error(
+		'SESSION_SECRET not found in environment vars, cannot continue!'
+	);
+	process.exit(1);
+}
+
+const router = express.Router();
+
+const stage = process.env.NODE_ENV;
```

Comment:
stage seems like a weird name, but then I'm not familiar with the terminology.

Reply:
Yeah it's an AWS thing 🤐 

They use "environment" to mean "region and account", whereas what we know as environments are called stages. Which is doubly odd when it's quite common to have a "staging" environment, so then you'd have "staging" stage 😕

Reply:
Ah well! Can't win them all